packages/transaction-preview/src/index.ts

```diff
@@ -137,6 +111,95 @@ export const patchProvider = (
   return patchedProvider
 }
 
+const handlePreview = async (
+  transaction: TransactionForSim[],
+  fullProviderRequest?: PatchedEIP1193Provider['request'],
+  req?: {
+    method: string
+    params?: Array<unknown>
+  }
+): Promise<void | unknown> => {
+  try {
+    if (!options) {
+      throw new Error(
+        `Please initialize Transaction Preview package prior to previewing a transaction.
+         You can do this by calling the init function with the appropriate params`
+      )
+    }
+    const preview = await simulateTransactions(options, transaction)
+    if (preview.error.length) {
```

Comment:
@Adamj1232 is there a case where `preview` | `error` could be undefined here?

Reply:
Good call! Added null check for preview above this block although it should always be there and if an error occurs is caught in the fetch